service/frontend/service.go

```diff
@@ -124,153 +125,117 @@ func NewConfig(dc *dynamicconfig.Collection, numHistoryShards int, enableReadFro
+		params:   params,
+	}, nil
+}
 
-	historyArchiverBootstrapContainer := &archiver.HistoryBootstrapContainer{
-		HistoryV2Manager: historyV2,
-		Logger:           base.GetLogger(),
-		MetricsClient:    base.GetMetricsClient(),
-		ClusterMetadata:  base.GetClusterMetadata(),
-		DomainCache:      domainCache,
-	}
-	visibilityArchiverBootstrapContainer := &archiver.VisibilityBootstrapContainer{
-		Logger:          base.GetLogger(),
-		MetricsClient:   base.GetMetricsClient(),
-		ClusterMetadata: base.GetClusterMetadata(),
-		DomainCache:     domainCache,
-	}
-	err = params.ArchiverProvider.RegisterBootstrapContainer(common.FrontendServiceName, historyArchiverBootstrapContainer, visibilityArchiverBootstrapContainer)
-	if err != nil {
-		log.Fatal("Failed to register archiver bootstrap container", tag.Error(err))
-	}
+// Start starts the service
+func (s *Service) Start() {
+
+	logger := s.GetLogger()
+	logger.Info("frontend starting", tag.Service(common.FrontendServiceName))
 
 	var replicationMessageSink messaging.Producer
-	var domainReplicationQueue persistence.DomainReplicationQueue
-	clusterMetadata := base.GetClusterMetadata()
+	clusterMetadata := s.GetClusterMetadata()
 	if clusterMetadata.IsGlobalDomainEnabled() {
 		consumerConfig := clusterMetadata.GetReplicationConsumerConfig()
 		if consumerConfig != nil && consumerConfig.Type == config.ReplicationConsumerTypeRPC {
-			domainReplicationQueue, err = pFactory.NewDomainReplicationQueue()
-			if err != nil {
-				log.Fatal("Failed to create domain replication queue", tag.Error(err))
-			}
-			replicationMessageSink = domainReplicationQueue
+			replicationMessageSink = s.GetDomainReplicationQueue()
 		} else {
-			replicationMessageSink, err = base.GetMessagingClient().NewProducerWithClusterName(
-				base.GetClusterMetadata().GetCurrentClusterName())
+			var err error
+			replicationMessageSink, err = s.GetMessagingClient().NewProducerWithClusterName(
+				s.GetClusterMetadata().GetCurrentClusterName())
 			if err != nil {
-				log.Fatal("Creating replicationMessageSink producer failed", tag.Error(err))
+				logger.Fatal("Creating replicationMessageSink producer failed", tag.Error(err))
 			}
 		}
 	} else {
 		replicationMessageSink = &mocks.KafkaProducer{}
+		replicationMessageSink.(*mocks.KafkaProducer).On("Publish", mock.Anything).Return(nil)
 	}
 
-	wfHandler := NewWorkflowHandler(
-		base,
-		s.config,
-		metadata,
-		historyV2,
-		visibility,
-		replicationMessageSink,
-		domainReplicationQueue,
-		domainCache)
-	dcRedirectionHandler := NewDCRedirectionHandler(wfHandler, params.DCRedirectionPolicy)
+	wfHandler := NewWorkflowHandler(s, s.config, replicationMessageSink)
+	dcRedirectionHandler := NewDCRedirectionHandler(wfHandler, s.params.DCRedirectionPolicy)
 	dcRedirectionHandler.RegisterHandler()
 
-	adminHandler := NewAdminHandler(base, pConfig.NumHistoryShards, domainCache, historyV2, s.params, s.config)
+	adminHandler := NewAdminHandler(s, s.params, s.config)
```

Comment:
plz do not make NewAdminHandler depending on params. passing in num of shards is ok

Reply:
Synced offline. This will be addressed latter as `params` is used multiple times in `adminHandler`